### PR TITLE
Migrate integratioon tests to typescript

### PR DESCRIPTION
**Pull request by birkskyum**
Now that no old .js code rely on our utility functions, we can move it all to typescript - and then a progressively remove remaining ts-ignore flags.

### CODE REVIEW

**Comment by wipfli** [comment]
Isn't `__dirname` already defined by default?

**Comment by wipfli** [comment]
Can you resolve the typescript ignore statements as part of this pull request?

**Comment by wipfli** [comment]
Other than that, this looks good to me. Thanks :)

**Comment by birkskyum** [comment]
I tried to remove it in the harness just now, and got this:

```
ReferenceError: __dirname is not defined
```

**Comment by birkskyum** [comment]
It's quite tough to get rid of all these ts-ignore in the existing scripts - but I am hopeful that we can get rid of most by replacing the harness etc. with a different test setup, and then the files they are in can be taken out.

**Comment by birkskyum** [comment]
I removed the ts-ignore from import.meta.url. My editor complain, but the code compiles and runs just fine.

File: test/integration/lib/server.ts [hunk 1]
```diff
@@ -6,14 +6,14 @@ import localizeURLs from './localize-urls.js';
...
-const require = createRequire(import.meta.url);
+const requireFn = createRequire(import.meta.url);
 const __dirname = dirname(fileURLToPath(import.meta.url));
 
 export default function () {
     const port = 2900;
     const integrationMount = st({path: path.join(__dirname, '..')});
-    const mapboxGLStylesMount = st({path: path.dirname(require.resolve('mapbox-gl-styles')), url: 'mapbox-gl-styles'});
-    const mapboxMVTFixturesMount = st({path: path.dirname(require.resolve('@mapbox/mvt-fixtures')), url: 'mvt-fixtures'});
+    const mapboxGLStylesMount = st({path: path.dirname(requireFn.resolve('mapbox-gl-styles')), url: 'mapbox-gl-styles'});
```
Comment (wipfli) [inline test/integration/lib/server.ts:15]: Can you change this to maplibre?

Comment (wipfli) [reply test/integration/lib/server.ts]: Or maybe not since it still is mapbox?

Comment (birkskyum) [reply test/integration/lib/server.ts]: Yes, it's maplibre now.

File: test/util/simulate_interaction.ts [hunk 2]
```diff
@@ -11,14 +11,15 @@ export function window(target) {
 
 const events = {};
 
+// @ts-ignore
 events.click = function (target, options) {
```
Comment (wipfli) [inline test/util/simulate_interaction.ts:15]: Does `events['click']` work?

File: test/integration/custom_layer_implementations.ts [hunk 3]
```diff
@@ -39,6 +43,12 @@ class NullIsland {
 }
 
 class Tent3D {
+    id: string;
+    type: string;
+    renderingMode: string;
+    program: any;
```
Comment (wipfli) [inline test/integration/custom_layer_implementations.ts:49]: Add better typings.

**Review by wipfli — CHANGES_REQUESTED** [review]

File: test/integration/lib/server.ts (hunk 1 above)
Comment (wipfli) [reply test/integration/lib/server.ts]: sounds good. thanks :)

File: test/integration/custom_layer_implementations.ts (hunk 3 above)
Comment (wipfli) [reply test/integration/custom_layer_implementations.ts:49]: I think it is `WebGLBuffer`

Comment (wipfli) [reply test/integration/custom_layer_implementations.ts:49]: https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bindBuffer

Comment (wipfli) [reply test/integration/custom_layer_implementations.ts:49]: And `WebGLProgram`: https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/attachShader

Comment (birkskyum) [reply test/integration/custom_layer_implementations.ts]: I added all the webgl types just now. For some reason we add unknown properties to WebGLProgram, like uMatrix, aPos, a_pos. I'm not sure if a_pos and aPos is a typo or intentional, but at least this move to typescript show us more clealy what is going on.

File: test/util/simulate_interaction.ts [hunk 4]
```diff
@@ -66,6 +66,7 @@ events.dblclick = function (target, options) {
 
 // magic deltaY value that indicates the event is from a mouse wheel
 // (rather than a trackpad)
+// @ts-ignore
 events.magicWheelZoomDelta = 4.000244140625;
```
Comment (wipfli) [inline test/util/simulate_interaction.ts:70]: Use `events['magicWheelZoomDelta']`

**Review by wipfli — APPROVED** [review]
Fix last ts-ignore and then this is approved. Great that you are working on this. Thanks a lot Birk.

File: test/integration/custom_layer_implementations.ts (hunk 3 above)
Comment (wipfli) [reply test/integration/custom_layer_implementations.ts]: Cool

**Pull request merged**